src/codelens/flinkSqlProvider.test.ts

```diff
@@ -225,3 +232,227 @@ describe("codelens/flinkSqlProvider.ts", () => {
+    getConfigStub.withArgs(FLINK_CONFIG_DATABASE).returns("some-other-database-id");
+    const metadata: UriMetadata = {
+      [UriMetadataKeys.FLINK_DATABASE_ID]: TEST_CCLOUD_KAFKA_CLUSTER.id,
+    };
+    const envs: CCloudEnvironment[] = [testFlinkEnv];
+    const catalogDb: CatalogDatabase = await getCatalogDatabaseFromMetadata(metadata, envs);
+
+    assert.strictEqual(catalogDb.catalog, testFlinkEnv);
+    assert.ok(catalogDb.database instanceof CCloudKafkaCluster);
+    assert.strictEqual(catalogDb.database.id, TEST_CCLOUD_KAFKA_CLUSTER.id);
+  });
+
+  it(`should return no catalog/database if the stored "${UriMetadataKeys.FLINK_DATABASE_ID}" metadata doesn't match any database`, async () => {
+    getConfigStub.withArgs(FLINK_CONFIG_DATABASE).returns(undefined);
+    const metadata: UriMetadata = {
+      [UriMetadataKeys.FLINK_DATABASE_ID]: "non-existent-database-id",
+    };
+    const envs: CCloudEnvironment[] = [testFlinkEnv];
+    const catalogDb: CatalogDatabase = await getCatalogDatabaseFromMetadata(metadata, envs);
+
+    assert.strictEqual(catalogDb.catalog, undefined);
+    assert.strictEqual(catalogDb.database, undefined);
+  });
+
+  it("should return database when compute pool is provided and database provider/region matches", async () => {
+    // provider/region match TEST_CCLOUD_KAFKA_CLUSTER by default
+    const matchingComputePool = TEST_CCLOUD_FLINK_COMPUTE_POOL;
+
+    getConfigStub.withArgs(FLINK_CONFIG_DATABASE).returns(undefined);
+    const metadata: UriMetadata = {
+      [UriMetadataKeys.FLINK_DATABASE_ID]: TEST_CCLOUD_KAFKA_CLUSTER.id,
+    };
+    const envs: CCloudEnvironment[] = [testFlinkEnv];
+    const catalogDb: CatalogDatabase = await getCatalogDatabaseFromMetadata(
+      metadata,
+      envs,
+      matchingComputePool,
+    );
+
+    assert.strictEqual(catalogDb.catalog, testFlinkEnv);
+    assert.ok(catalogDb.database instanceof CCloudKafkaCluster);
+    assert.strictEqual(catalogDb.database.id, TEST_CCLOUD_KAFKA_CLUSTER.id);
+  });
+
+  it("should return catalog but no database when compute pool is provided but database provider/region doesn't match", async () => {
+    // compute pool that doesn't match TEST_CCLOUD_KAFKA_CLUSTER's provider/region
+    const nonMatchingComputePool = new CCloudFlinkComputePool({
+      ...TEST_CCLOUD_FLINK_COMPUTE_POOL,
+      provider: "FAKE-PROVIDER",
+      region: "abc-fakeregion-999",
+    });
+
+    getConfigStub.withArgs(FLINK_CONFIG_DATABASE).returns(undefined);
+    const metadata: UriMetadata = {
+      [UriMetadataKeys.FLINK_DATABASE_ID]: TEST_CCLOUD_KAFKA_CLUSTER.id,
+    };
+    const envs: CCloudEnvironment[] = [testFlinkEnv];
+    const catalogDb: CatalogDatabase = await getCatalogDatabaseFromMetadata(
+      metadata,
+      envs,
+      nonMatchingComputePool,
+    );
+
+    assert.strictEqual(catalogDb.catalog, testFlinkEnv);
+    assert.strictEqual(catalogDb.database, undefined);
```

Comment:
Having a `catalog` but not a `database` currently doesn't affect anything since the codelenses require both in order to display information:
https://github.com/confluentinc/vscode/blob/6b1e5113f92f1901623214432e432bd9844a16fb/src/codelens/flinkSqlProvider.ts#L105-L111